Fix user based api calls and forms.

johndavis92790 opened the pull request:
- Fix login and sign-up pages.
- Add login nevbar option when no user is logged in, change to logout when the user has logged in.
- Fix user based api calls with mongoDB.
- Refactor component and page function calls for more consistency across the app.
- Create useStates for currentUserEmail and currentUser to keep track of current user after logging in and to be able to access the user information from anywhere within the app.

johndavis92790 committed:
Fix user based api calls and forms.
- Fix login and sign-up pages.
- Add login nevbar option when no user is logged in, change to logout when the user has logged in.
- Fix user based api calls with mongoDB.
- Refactor component and page function calls for more consistency across the app.
- Create useStates for currentUserEmail and currentUser to keep track of current user after logging in and to be able to access the user information from anywhere within the app.

robyng commented on client/src/App.tsx:
```diff
@@ -1,30 +1,34 @@
       const content = await response.json();
-      setCurrentUserEmail(content.username);
+      setCurrentUserEmail(content.email);
+      setCurrentUser(content);
```
does this need to be content.username?

johndavis92790 replied:
Do we want to have users login with a username or email? I can set it up either way, you choose

johndavis92790 replied:
I remember now, I used email because thats how David setup the login page, but we can change that. You let me know how you want it setup and I will take care of it.

robyng replied:
Okay I see what you are saying where Login has email field, we should keep it as that then. Future development would be great for the field to accept either email or username to login.

robyng commented on client/src/utils/helpers.tsx:
```diff
@@ -1,23 +1,19 @@
-  } else {
+  if (slice === "http://localhost:3000") {
+    return "http://localhost:3001";
+  } else if (slice === "https://dragon-fruit-") {
```
Should the slice be https://dragon-fruit-app ?

johndavis92790 replied:
No because this has to be the first 22 characters. This is kind of a hacky way to get the API to work in both local and live environments. I will eventually find a better, less hacky way, but this was super easy for now and works fine.

robyng replied:
Okay! I see you this is how your identifying live or local environment for the API. Thanks. If we ever change our domain name we would need to update this here.

robyng reviewed (changes requested):
Alright doing export const... from now on! Just a few questions that I had.

robyng committed:
Merge branch 'develop' into john/user-security

robyng committed:
npm install.

Prettier changes.

Add back RestaurantsResults to app.tsx

Also fix import link to GooglePhoto in SingleSearchResult.tsx

robyng committed:
add sign up text back

robyng committed:
Nest MainContainer in form element to fix style.

robyng commented:
Some funky things happening when I pull down and test and try sign up or login. Console shows endless errors of user unauthorized and doesn't show token in applications. Maybe I'm missing something?

robyng committed:
Remove SignUp style.

Swap order of Form and MainContainer to keep previous styles

robyng commented on server/controller/userController.js:
```diff
@@ -1,10 +1,10 @@
-const axios = require("axios");
+// const axios = require('axios');
 const bcrypt = require("bcryptjs");
 const jwt = require("jsonwebtoken");
```
Does a web token show when you login? No tokens are showing for me when I test.

johndavis92790 replied:
If you haven't yet, please update your .env file, maybe that will fix this. I sent you the info on Slack.

johndavis92790 committed:
Fix currentUser useState in app.tsx.

johndavis92790 committed:
Merge branch 'develop' of https://github.com/Healthy-Plate-Project/Healthy-Eating-App into john/user-security

johndavis92790 committed:
Merge branch 'john/user-security' of https://github.com/Healthy-Plate-Project/Healthy-Eating-App into john/user-security

johndavis92790 committed:
Update logic for login, logout proccesses.

johndavis92790 committed:
Add id to UserData object.
- Also update userController to send the id in its response.

robyng commented:
Thanks, seeing token in the cookies. Would like to have the sign up be able to login automatically instead of sing up  then login. Will make a feature development ticket for it.

johndavis92790 commented:
> Thanks, seeing token in the cookies. Would like to have the sign up be able to login automatically instead of sing up then login. Will make a feature development ticket for it.

Ya thats fine, this is the simpler approach so to save time I did it this way but we can change it later no problem.
